regolith/builders/beamplanbuilder.py

```diff
@@ -0,0 +1,98 @@
+"""Builder for Current and Pending Reports."""
+from regolith.builders.basebuilder import LatexBuilderBase
+from regolith.tools import all_docs_from_collection
+from regolith.tools import group
+import pandas as pd
+
+
+class BeamPlanBuilder(LatexBuilderBase):
+    """
+    Build a file of experiment plans for the beamtime from database entries. The report is in .tex file. The template
+    of the file is in the 'templates/beamplan.txt'. The data will be grouped according to beamtime. Each beamtime
+    will generate a file of the plans. If 'beamtime' in 'rc' are not None, only plans for those beamtime will be
+    generated.
+    """
+    btype = "beamplan"
+    needed_dbs = ['beamplan', "people"]
+
+    def construct_global_ctx(self):
+        """Constructs the global context"""
+        super().construct_global_ctx()
+        gtx = self.gtx
+        rc = self.rc
+        gtx["beamplan"] = all_docs_from_collection(rc.client, "beamplan")
+        gtx["all_docs_from_collection"] = all_docs_from_collection
+        gtx["float"] = float
+        gtx["str"] = str
+
+    @staticmethod
+    def gather_info(docs):
+        """
+        Make a table as the summary of the plans and a list of experiment plans. The table header contains: serial
+        id, person name, number of sample, sample container, sample holder, measurement, estimated time (min). The
```

Comment:
we don't need this much detail about what the report looks like, the person can look at the template.

Reply:
it will be abbreviated.